### DISCUSSION TITLE
GitHub Copilot is too fast

### DISCUSSION THREAD

**Discussion by Supermarcel10** [community/community#122181] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Copilot used to be slower, meaning that it wasn't as annoying when writing code. However, with recent updates it suggests code a lot faster, making it nearly impossible to ignored the code at times, and it will continue suggesting when the user types.

Worst of all, is that these suggestions are utterly useless, mostly incorrect and often just make the workflow longer. For this reason I've decided to stop using Copilot, because it ends up slowing me down, rather than helping in any meaningful way.

**Comment by bradezard131** [comment]
I also find the increased eagerness problematic. It triggers lag in the VSCode UI which is distracting and unpleasant. To be a little more constructive, it would be great if we could have a configurable delay before inline completion triggered. There's already discussions about the "Copilot Pause" (when developers using Copilot will pause for a moment after typing some code, waiting for Copilot to complete it), letting us double-down on this effect as an intended usage pattern would be very helpful. I'm sure that I and many others have an instinct for the types of prefixes that work well for Copilot for us, we know how fast we type too, and so can dial in how long of a pause should be required to trigger completions.

**Comment by tgrushka** [comment]
Yep, a year old issue for a for-profit product. Copilot is not open source. It's not free. Why is there no option for a delay in the settings? Completions are either enabled or disabled. You can't shut it up without disabling it. Then later you wonder why it's not working at all -- duh, you have to re-enable it. All it would take is adding a setting to the extension config and several lines of TypeScript code. In fact GitHub could have their own Copilot help them implement this fix.

**Comment by mindcrime** [comment]
Yes please. Just add a setting like "don't suggest unless the user has not typed a character for _n_ milliseconds" or something. Or a keybinding to a "pause suggestions for _n_ milliseconds" action. Or both.

**Comment by rahulacaleffi** [comment]
Agreed 100%. It's being too fast. Bothers me, 'cause interrupts our thinking line.

**Comment by advanced03** [comment]
Agreed

**Comment by bradezard131** [comment]
It appears this is finally in the latest August release `editor.inlineSuggest.minShowDelay`. This will legitimately improve my experience so much, I can't believe it took over a year to add a sleep call.

**Reply by AdamReece-WebBox** [reply]
Can confirm. I set this to `1500` for 1.5 seconds, seems to work well.

**Reply by advanced03** [reply]
> It appears this is finally in the latest August release `editor.inlineSuggest.minShowDelay`. This will legitimately improve my experience so much, I can't believe it took over a year to add a sleep call.

Really? Let me try this!
